Parquet derive fails to build when Result is aliased

**Describe the bug**
Right now 

**To Reproduce**

Attempting to compile something where `Result` is aliased to something with a different signature (which I feel is a common pattern), fails to compile bc the proc macro generates code that uses that type, rather than the `std::result::Result`. For example this code: 

```rust
use parquet_derive::ParquetRecordReader;

#[derive(Debug)]
enum MyInternalError {
    IForgotSomething,
    ExpectecNotEmpty,
}

pub type Result = std::result::Result<(), MyInternalError>;

#[derive(Debug, ParquetRecordReader)]
struct MyRecord {
    name: String,
    age: u32,
}

fn main() {
    println!("Hello, world!");
}

```

Fails to compile (`cargo b`) with this error (minor edits to remove some paths...)

```
error[E0107]: type alias takes 0 generic arguments but 2 generic arguments were supplied
  --> src/main.rs:13:17
   |
13 | #[derive(Debug, ParquetRecordReader)]
   |                 ^^^^^^^^^^^^^^^^^^^ expected 0 generic arguments
   |
note: type alias defined here, with 0 generic parameters
  --> src/main.rs:11:10
   |
11 | pub type Result = std::result::Result<(), MyInternalError>;
   |          ^^^^^^
   = note: this error originates in the derive macro `ParquetRecordReader` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0053]: method `read_from_row_group` has an incompatible type for trait
  --> src/main.rs:13:17
   |
13 | #[derive(Debug, ParquetRecordReader)]
   |                 ^^^^^^^^^^^^^^^^^^^ expected `ParquetError`, found `MyInternalError`
   |
   = note: expected signature `fn(&mut Vec<_>, &mut dyn RowGroupReader, _) -> std::result::Result<_, ParquetError>`
              found signature `fn(&mut Vec<_>, &mut dyn RowGroupReader, _) -> std::result::Result<_, MyInternalError>`
   = note: this error originates in the derive macro `ParquetRecordReader` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0308]: mismatched types
   --> src/main.rs:13:17
    |
13  | #[derive(Debug, ParquetRecordReader)]
    |                 ^^^^^^^^^^^^^^^^^^^
    |                 |
    |                 expected `MyInternalError`, found `ParquetError`
    |                 arguments to this enum variant are incorrect
    |
help: the type constructed contains `ParquetError` due to the type of the argument passed
   --> src/main.rs:13:17
    |
13  | #[derive(Debug, ParquetRecordReader)]
    |                 ^^^^^^^^^^^^^^^^^^^ this argument influences the type of `Err`
note: tuple variant defined here
   --> $HOME/.rustup/toolchains/stable-aarch64-apple-darwin/lib/rustlib/src/rust/library/core/src/result.rs:537:5
    |
537 |     Err(#[stable(feature = "rust1", since = "1.0.0")] E),
    |     ^^^
    = note: this error originates in the derive macro `ParquetRecordReader` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: `?` couldn't convert the error to `MyInternalError`
  --> src/main.rs:13:35
   |
13 | #[derive(Debug, ParquetRecordReader)]
   |                 ------------------^
   |                 |                 |
   |                 |                 the trait `From<ParquetError>` is not implemented for `MyInternalError`
   |                 this can't be annotated with `?` because it has type `Result<_, ParquetError>`
   |
   = note: the question mark operation (`?`) implicitly performs a conversion on the error value using the `From` trait
   = help: the trait `FromResidual<std::result::Result<Infallible, E>>` is implemented for `std::result::Result<T, F>`
   = note: required for `std::result::Result<(), MyInternalError>` to implement `FromResidual<std::result::Result<Infallible, ParquetError>>`
   = note: this error originates in the derive macro `ParquetRecordReader` (in Nightly builds, run with -Z macro-backtrace for more info)

Some errors have detailed explanations: E0053, E0107, E0277, E0308.
For more information about an error, try `rustc --explain E0053`.
error: could not compile `parquet_derive_test` (bin "parquet_derive_test") due to 4 previous errors

```

**Expected behavior**
Succesful compilation.

**Additional context**
None that I am aware of.

--- a/parquet_derive/src/lib.rs
+++ b/parquet_derive/src/lib.rs
@@ -114,7 +114,7 @@ pub fn parquet_record_writer(input: proc_macro::TokenStream) -> proc_macro::Toke
       fn write_to_row_group<W: ::std::io::Write + Send>(
         &self,
         row_group_writer: &mut ::parquet::file::writer::SerializedRowGroupWriter<'_, W>
-      ) -> std::result::Result<(), ::parquet::errors::ParquetError> {
+      ) -> ::std::result::Result<(), ::parquet::errors::ParquetError> {
         use ::parquet::column::writer::ColumnWriter;
 
         let mut row_group_writer = row_group_writer;
@@ -135,7 +135,7 @@ pub fn parquet_record_writer(input: proc_macro::TokenStream) -> proc_macro::Toke
         Ok(())
       }
 
-      fn schema(&self) -> std::result::Result<::parquet::schema::types::TypePtr, ::parquet::errors::ParquetError> {
+      fn schema(&self) -> ::std::result::Result<::parquet::schema::types::TypePtr, ::parquet::errors::ParquetError> {
         use ::parquet::schema::types::Type as ParquetType;
         use ::parquet::schema::types::TypePtr;
         use ::parquet::basic::LogicalType;
@@ -211,7 +211,7 @@ pub fn parquet_record_reader(input: proc_macro::TokenStream) -> proc_macro::Toke
         &mut self,
         row_group_reader: &mut dyn ::parquet::file::reader::RowGroupReader,
         num_records: usize,
-      ) -> std::result::Result<(), ::parquet::errors::ParquetError> {
+      ) -> ::std::result::Result<(), ::parquet::errors::ParquetError> {
         use ::parquet::column::reader::ColumnReader;
 
         let mut row_group_reader = row_group_reader;

--- a/parquet_derive/src/lib.rs
+++ b/parquet_derive/src/lib.rs
@@ -114,7 +114,7 @@ pub fn parquet_record_writer(input: proc_macro::TokenStream) -> proc_macro::Toke
       fn write_to_row_group<W: ::std::io::Write + Send>(
         &self,
         row_group_writer: &mut ::parquet::file::writer::SerializedRowGroupWriter<'_, W>
-      ) -> Result<(), ::parquet::errors::ParquetError> {
+      ) -> std::result::Result<(), ::parquet::errors::ParquetError> {
         use ::parquet::column::writer::ColumnWriter;
 
         let mut row_group_writer = row_group_writer;
@@ -135,7 +135,7 @@ pub fn parquet_record_writer(input: proc_macro::TokenStream) -> proc_macro::Toke
         Ok(())
       }
 
-      fn schema(&self) -> Result<::parquet::schema::types::TypePtr, ::parquet::errors::ParquetError> {
+      fn schema(&self) -> std::result::Result<::parquet::schema::types::TypePtr, ::parquet::errors::ParquetError> {
         use ::parquet::schema::types::Type as ParquetType;
         use ::parquet::schema::types::TypePtr;
         use ::parquet::basic::LogicalType;
@@ -211,7 +211,7 @@ pub fn parquet_record_reader(input: proc_macro::TokenStream) -> proc_macro::Toke
         &mut self,
         row_group_reader: &mut dyn ::parquet::file::reader::RowGroupReader,
         num_records: usize,
-      ) -> Result<(), ::parquet::errors::ParquetError> {
+      ) -> std::result::Result<(), ::parquet::errors::ParquetError> {
         use ::parquet::column::reader::ColumnReader;
 
         let mut row_group_reader = row_group_reader;

--- a/parquet_derive_test/src/lib.rs
+++ b/parquet_derive_test/src/lib.rs
@@ -347,6 +347,36 @@ mod tests {
         assert_eq!(drs[0].isize, out[0].isize);
     }
 
+    #[test]
+    fn test_aliased_result() {
+        // Issue 7547, Where aliasing the `Result` led to
+        // a collision with the macro internals of derive ParquetRecordReader
+
+        mod aliased_result {
+            use parquet_derive::{ParquetRecordReader, ParquetRecordWriter};
+            pub type Result = std::result::Result<(), Box<dyn std::error::Error>>;
+
+            #[derive(ParquetRecordReader, ParquetRecordWriter, Debug)]
+            pub struct ARecord {
+                pub bool: bool,
+                pub string: String,
+            }
+
+            impl ARecord {
+                pub fn validate(&self) -> Result {
+                    Ok(())
+                }
+            }
+        }
+
+        use aliased_result::ARecord;
+        let foo = ARecord {
+            bool: true,
+            string: "test".to_string(),
+        };
+        assert!(foo.validate().is_ok());
+    }
+
     /// Returns file handle for a temp file in 'target' directory with a provided content
     pub fn get_temp_file(file_name: &str, content: &[u8]) -> fs::File {
         // build tmp path to a file in "target/debug/testdata"

--- a/parquet_derive_test/src/lib.rs
+++ b/parquet_derive_test/src/lib.rs
@@ -354,7 +354,13 @@ mod tests {
 
         mod aliased_result {
             use parquet_derive::{ParquetRecordReader, ParquetRecordWriter};
-            pub type Result = std::result::Result<(), Box<dyn std::error::Error>>;
+
+            // This is the normal pattern that raised this issue
+            // pub type Result = std::result::Result<(), Box<dyn std::error::Error>>;
+
+            // not an actual result type
+            // Used here only to make the harder.
+            pub type Result = ();
 
             #[derive(ParquetRecordReader, ParquetRecordWriter, Debug)]
             pub struct ARecord {
@@ -363,7 +369,8 @@ mod tests {
             }
 
             impl ARecord {
-                pub fn validate(&self) -> Result {
+                pub fn do_nothing(&self) -> Result {}
+                pub fn validate(&self) -> std::result::Result<(), Box<dyn std::error::Error>> {
                     Ok(())
                 }
             }
@@ -374,6 +381,7 @@ mod tests {
             bool: true,
             string: "test".to_string(),
         };
+        foo.do_nothing();
         assert!(foo.validate().is_ok());
     }
 

